### PR TITLE
adding terms for chairs

### PR DESCRIPTION
**Pull request by parispittman**
for discussion; split from [terms and term limits issue](https://github.com/kubernetes/community/issues/5886)

notes - 
- this is not the technical lead role
- some links are missing

### CODE REVIEW

File: committee-steering/governance/sig-governance.md [hunk 1]
```diff
@@ -82,7 +82,21 @@ Subproject contributors (as applicable).
...
 ### Chair
 
+From governance.md:
+> SIG Chairs are intended to be organizers and facilitators, responsible for 
+> the operation of the SIG and for communication and coordination with the 
+> other SIGs, the Steering Committee, and the broader community. 
+
 - Number: 2-3
+- Term: 3 years
+  - 90 days before term is up, Chair will send an email to the SIG mailing list,
```
Comment (detiber) [inline committee-steering/governance/sig-governance.md:92]: Normally I would be against putting work on external groups, but considering responsiveness of chairs has been an issue in the past, would it be good for this to be triggered by an external group such as Steering or SIG ContribEx/Elections Subproject to track the term start/end period and triggering this announcement?

File: committee-steering/governance/sig-governance.md [hunk 2]
```diff
@@ -82,7 +82,21 @@ Subproject contributors (as applicable).
...
+    announce at the regularly scheduled meeting, and send a note to the slack
+    channel asking for interested individuals to step forward. This provides
+    time for outreach, training, and onboarding. 
+  - If no one steps up in 90 days, the current Chair can assume another one (1)
+    year position as Chair while looking for a replacement.
+  Note: The Chair could rotate to Technical Lead (TL) if other TLs agree
```
Comment (detiber) [inline committee-steering/governance/sig-governance.md:98]: Maybe reword this to: `Note: This would not prevent the Chair from assuming another Lead role (other than Chair) as long as all usual processes are followed.`

File: committee-steering/governance/sig-governance.md (hunk 1 above)
Comment (detiber) [reply committee-steering/governance/sig-governance.md:92]: I'm wondering if it would make sense to also add a statement to the effect of: Chair terms should be staggered for continuity purposes

Comment (detiber) [reply committee-steering/governance/sig-governance.md:92]: For existing Chairs, should the 3 year term start from when they started as chair or from when the new policy takes effect? In either case, there should likely be some adjustments made for ensuring all current leads do not have their terms end at the same time.

I'm also now wondering if it would make sense to have the terms applied individually or if it would make sense to define the terms based on calendar year (and possibly staggered by SIG). Individually seems like it would be a lot harder to keep track of, where based on calendar year it could more easily be published and available for the community to see. The use of calendar year would also make it easier to leverage the [Elections Subproject](https://github.com/kubernetes/community/tree/master/elections/teams) for the voting process, though I'm not quite sure if Electo supports a differentiation between binding/non-binding votes as defined by the On-boarding and Off-barding Leads section above.

Comment (parispittman) [reply committee-steering/governance/sig-governance.md:92]: the last comment is a good discussion topic that I was hoping to have with the wider group. 

the first comment - we can add in month and date in sigs.yaml to help track and contribex could indeed help but should be the responsibility of the sig first and foremost.

File: committee-steering/governance/sig-governance.md (hunk 2 above)
Comment (parispittman) [reply committee-steering/governance/sig-governance.md:98]: I'm deleting the note entirely

**Comment by aojea** [comment]
I find that this is not addressing the real problem, that is "Quis custodiet ipsos custodes?" and it is a solomonic solution instead of addressing the real problem.

If there are some SIGs that have Chairs that are not able to dedicate enough time to grow people and keep the SIG healthy, those should recognize the problem and look for help or for a succession, the problem is that this is not happening .... 

how do we solve this? I do not know, but I do not think that we should use a bandaid like this, because I think that there is consensus that the biggest problem right now is the lack of maintainers, and it requires months (optimistically) or years to grow people ... and these people need mentors

File: committee-steering/governance/sig-governance.md (hunk 1 above)
Comment (logicalhan) [reply committee-steering/governance/sig-governance.md:92]: Does this mean chairs have a 1 term limit?

File: committee-steering/governance/sig-governance.md [hunk 3]
```diff
@@ -82,9 +82,22 @@ Subproject contributors (as applicable).
...
+> SIG Chairs are intended to be organizers and facilitators, responsible for 
+> the operation of the SIG and for communication and coordination with the 
+> other SIGs, the Steering Committee, and the broader community. 
+
 - Number: 2-3
+- Term: 3 years
+  - 90 days before term is up, Chair will send an email to the SIG mailing list,
+    announce at the regularly scheduled meeting, and send a note to the slack
+    channel asking for interested individuals to step forward. This provides
+    time for outreach, training, and onboarding.
```
Comment (ahg-g) [inline committee-steering/governance/sig-governance.md:95]: what is the process if more than one candidate steps up?

Comment (logicalhan) [reply committee-steering/governance/sig-governance.md:95]: In sig instrumentation we have used cordecet voting.

File: committee-steering/governance/sig-governance.md (hunk 1 above)
Comment (parispittman) [reply committee-steering/governance/sig-governance.md:92]: @logicalhan limits up for discussion at this point; what do you think?

Comment (logicalhan) [reply committee-steering/governance/sig-governance.md:92]: Currently worded, it almost seems implied that `term limit == 1`. I am personally for terms (as I believe in the democratic process). Term limits may end up being necessary too, but I think that (1) warrants a different thread/PR, (2) should probably be decided some time after implementing terms.

File: committee-steering/governance/sig-governance.md [hunk 4]
```diff
@@ -82,9 +82,22 @@ Subproject contributors (as applicable).
...
+> the operation of the SIG and for communication and coordination with the 
+> other SIGs, the Steering Committee, and the broader community. 
+
 - Number: 2-3
+- Term: 3 years
+  - 90 days before term is up, Chair will send an email to the SIG mailing list,
+    announce at the regularly scheduled meeting, and send a note to the slack
+    channel asking for interested individuals to step forward. This provides
+    time for outreach, training, and onboarding. 
+  - If no one steps up in 90 days, the current Chair can assume another one (1)
```
Comment (lavalamp) [inline committee-steering/governance/sig-governance.md:96]: It would be simpler to have all terms be the same length (I suggest 2y), and permit multiple terms (I suggest unlimited)

File: committee-steering/governance/sig-governance.md (hunk 3 above)
Comment (lavalamp) [reply committee-steering/governance/sig-governance.md:95]: Voting systems have to define who has standing to vote, how did you decide that?

Comment (logicalhan) [reply committee-steering/governance/sig-governance.md:95]: We try to keep [this up-to-date](https://github.com/kubernetes/org/blob/main/config/kubernetes/sig-instrumentation/teams.yaml#L21-L43).

File: committee-steering/governance/sig-governance.md (hunk 1 above)
Comment (vincepri) [reply committee-steering/governance/sig-governance.md:92]: > For existing Chairs, should the 3 year term start from when they started as chair or from when the new policy takes effect? In either case, there should likely be some adjustments made for ensuring all current leads do not have their terms end at the same time.

To strive more diversity of thought and leadership starting from the actual start date as chair would make sense. 

> limits up for discussion at this point; what do you think?

Just chiming in from my view in SIG Cluster Lifecycle, limits should be a forcing function to promote better community building, sharing of responsibilities, and generally more diversity of perspectives and thoughts. 

Limits for chairs (and perhaps TLs as well? although maybe a different discussion) can be both in terms of consecutive terms and total, an example we could have a limit of 1 consecutive term, and 5(ish?) total. There should be room for exceptions, like if nobody steps up for the role that a current chair can retain their role for a consecutive term, until a replacement is found.

File: committee-steering/governance/sig-governance.md (hunk 3 above)
Comment (divya-mohan0209) [reply committee-steering/governance/sig-governance.md:95]: By membership, I assume you mean active contributors/reviewers @logicalhan? For SIGs like ContribEx, Docs, and Release, this approach might not feasibly scale, IMO. If we choose to go down the voting route, I assume we'd need to refer to Devstats data just like we do with Steering elections.

File: committee-steering/governance/sig-governance.md (hunk 1 above)
Comment (divya-mohan0209) [reply committee-steering/governance/sig-governance.md:92]: We should consider mandating the number of chairs per SIG or modifying the terms & provisions for re-staffing SIGs with only one chair. This is because, for whatever reason, if the single chair can't continue to stay on for the one-year emeritus term while finding a replacement, there is a risk associated, and we need to account for that.

File: committee-steering/governance/sig-governance.md (hunk 3 above)
Comment (logicalhan) [reply committee-steering/governance/sig-governance.md:95]: I mean, we use devstats to populate our lists, so effectively, they are kinda the same thing.
